psi4/driver/driver.py

```diff
@@ -873,6 +873,194 @@ def properties(*args, **kwargs):
+    
+            self.p4_name = p4_name
+            self.p4_mol = p4_mol
+            self.p4_return_wfn = p4_return_wfn
+            self.p4_kwargs = p4_kwargs
+    
+            molecule = geometric.molecule.Molecule()
+            molecule.elem = [p4_mol.symbol(i) for i in range(p4_mol.natom())]
+            molecule.xyzs = [p4_mol.geometry().np * qcel.constants.bohr2angstroms] 
+            molecule.build_bonds()
+                                 
+            super(Psi4NativeEngine, self).__init__(molecule)
+    
+        def calc(self, coords, dirname):
+            self.p4_mol.set_geometry(core.Matrix.from_array(coords.reshape(-1,3)))
+            self.p4_mol.update_geometry()
+            if self.p4_return_wfn:
+                g, wfn = gradient(self.p4_name, return_wfn=True, molecule=self.p4_mol, **self.p4_kwargs)
+                self.p4_wfn = wfn
+            else:
+                g = gradient(self.p4_name, return_wfn=False, molecule=self.p4_mol, **self.p4_kwargs)
+            e = core.variable('CURRENT ENERGY')
+            return {'energy': e, 'gradient': g.np.ravel()}
+
+    return_wfn = kwargs.pop('return_wfn', False)
+    return_history = kwargs.pop('return_history', False)
+
+    if return_history:
+        step_energies = []
+        step_gradients = []
+        step_coordinates = []
+
+    # Make sure the molecule the user provided is the active one
+    molecule = kwargs.get('molecule', core.get_active_molecule())
+
+    # Do not fix orientation or COM
+    molecule.fix_orientation(True)
+    molecule.fix_com(True)
+    molecule.update_geometry()
+
+    # Get geometric-specific options
+    geometric_opts = kwargs.get('geometric_opts', {})
+
+    core.print_out('\n')
+    core.print_out("\n  ==> GeomeTRIC Optimizer <==                                                                   ~\n")
+                                 
+    # Default to Psi4 maxiter unless overridden
+    if 'maxiter' not in geometric_opts:
+        geometric_opts['maxiter'] = core.get_global_option('GEOM_MAXITER')
+
+    # Default to Psi4 geometry convergence criteria unless overridden 
+    if 'convergence_set' not in geometric_opts:
+        geometric_opts['convergence_set'] = core.get_global_option('G_CONVERGENCE')
+
+        # GeomeTRIC doesn't know these convergence criterion
+        if geometric_opts['convergence_set'] in ['CFOUR', 'QCHEM', 'MOLPRO']:
+            core.print_out(f"\n  Psi4 convergence criteria {geometric_opts['convergence_set']:6s} not recognized by GeomeTRIC, switching to GAU_TIGHT          ~")
+            geometric_opts['convergence_set'] = 'GAU_TIGHT'
+
+    engine = Psi4NativeEngine(name, molecule, return_wfn, **kwargs)
+    M = engine.M
+    
+    # Handle constraints
+    constraints_dict = geometric_opts.get('constraints', {})
+    constraints_string = geometric.run_json.make_constraints_string(constraints_dict)
+    Cons, CVals = None, None
+    if constraints_string:
+        if 'scan' in constraints_dict:
```

Comment:
Same here - just want to check that we know `scan` is always lower case

Reply:
Good catch!